singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt

```diff
@@ -9,37 +9,38 @@ import com.web3auth.singlefactorauth.types.AggregateVerifierParams
 import com.web3auth.singlefactorauth.types.ErrorCode
 import com.web3auth.singlefactorauth.types.LoginParams
 import com.web3auth.singlefactorauth.types.SFAError
-import com.web3auth.singlefactorauth.types.SingleFactorAuthArgs
-import com.web3auth.singlefactorauth.types.TorusKey
+import com.web3auth.singlefactorauth.types.SFAKey
+import com.web3auth.singlefactorauth.types.SFAParams
 import com.web3auth.singlefactorauth.types.TorusSubVerifierInfo
 import org.json.JSONObject
 import org.torusresearch.fetchnodedetails.FetchNodeDetails
 import org.torusresearch.fetchnodedetails.types.NodeDetails
 import org.torusresearch.torusutils.TorusUtils
-import org.torusresearch.torusutils.types.RetrieveSharesResponse
-import org.torusresearch.torusutils.types.TorusCtorOptions
-import org.torusresearch.torusutils.types.TorusPublicKey
-import org.torusresearch.torusutils.types.TypeOfUser
-import org.torusresearch.torusutils.types.VerifierArgs
+import org.torusresearch.torusutils.helpers.MetadataUtils
+import org.torusresearch.torusutils.types.VerifierParams
+import org.torusresearch.torusutils.types.VerifyParams
+import org.torusresearch.torusutils.types.common.TorusKey
+import org.torusresearch.torusutils.types.common.TorusOptions
+import org.torusresearch.torusutils.types.common.TorusPublicKey
+import org.torusresearch.torusutils.types.common.TypeOfUser
 import org.web3j.crypto.Hash
+import java.math.BigInteger
 import java.util.concurrent.CompletableFuture
 import java.util.concurrent.ExecutionException
 
-class SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs) {
-    private var nodeDetailManager: FetchNodeDetails
+class SingleFactorAuth(sfaParams: SFAParams) {
+    private var nodeDetailManager: FetchNodeDetails =
+        FetchNodeDetails(sfaParams.getNetwork())
     private val torusUtils: TorusUtils
     private lateinit var sessionManager: SessionManager
     private val gson = GsonBuilder().disableHtmlEscaping().create()
+    private var _sfaParams = sfaParams
 
     init {
-        nodeDetailManager = FetchNodeDetails(singleFactorAuthArgs.getNetwork())
-        val opts = TorusCtorOptions("single-factor-auth-android", singleFactorAuthArgs.clientid)
-        opts.isEnableOneKey = true
-        opts.network = singleFactorAuthArgs.getNetwork().toString()
-        opts.signerHost =
-            SingleFactorAuthArgs.SIGNER_MAP[singleFactorAuthArgs.getNetwork()] + "/api/sign"
-        opts.allowHost =
-            SingleFactorAuthArgs.SIGNER_MAP[singleFactorAuthArgs.getNetwork()] + "/api/allow"
+        val opts = TorusOptions(
```

Comment:
You're missing a parameter here from singlefactorauthargs, because SFAParams is in turn missing  a parameter in it's constructor.